core/src/main/java/io/questdb/std/histogram/org/HdrHistogram/ConcurrentHistogram.java

```diff
@@ -0,0 +1,691 @@
+/**
+ * Written by Gil Tene of Azul Systems, and released to the public domain,
+ * as explained at http://creativecommons.org/publicdomain/zero/1.0/
+ *
+ * @author Gil Tene
+ */
+
+package io.questdb.std.histogram.org.HdrHistogram;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicLongArray;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
+import java.util.zip.DataFormatException;
+
+/**
+ * An integer values High Dynamic Range (HDR) Histogram that supports safe concurrent recording operations.
+ * A ConcurrentHistogram guarantees lossless recording of values into the histogram even when the
+ * histogram is updated by multiple threads, and supports auto-resize and shift operations that may
+ * result from or occur concurrently with other recording operations.
+ * <p>
+ * It is important to note that concurrent recording, auto-sizing, and value shifting are the only thread-safe
+ * behaviors provided by {@link ConcurrentHistogram}, and that it is not otherwise synchronized. Specifically, {@link
+ * ConcurrentHistogram} provides no implicit synchronization that would prevent the contents of the histogram
+ * from changing during queries, iterations, copies, or addition operations on the histogram. Callers wishing to make
+ * potentially concurrent, multi-threaded updates that would safely work in the presence of queries, copies, or
+ * additions of histogram objects should either take care to externally synchronize and/or order their access,
+ * use the SynchronizedHistogram variant, or (recommended) use Recorder or
+ * SingleWriterRecorder which are intended for this purpose.
+ * <p>
+ * Auto-resizing: When constructed with no specified value range range (or when auto-resize is turned on with {@link
+ * Histogram#setAutoResize}) a {@link Histogram} will auto-resize its dynamic range to include recorded values as
+ * they are encountered. Note that recording calls that cause auto-resizing may take longer to execute, as resizing
+ * incurs allocation and copying of internal data structures.
+ * <p>
+ * See package description org.HdrHistogram for details.
+ */
+
+@SuppressWarnings("unused")
+public class ConcurrentHistogram extends Histogram {
```

Comment:
We should keep only the classes that are in use. No need to retain everything that was in the original library.